### ISSUE TITLE
 IconButton seems to be missing Semantics(container: true)

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#49119]
Internal bug: b/147830016

Comparing [RawMaterialButton](https://github.com/flutter/flutter/blob/8665e13801fe6b53a0e67866d2ee6cd5ef2083f4/packages/flutter/lib/src/material/button.dart#L458) to [IconButton](https://github.com/flutter/flutter/blob/8665e13801fe6b53a0e67866d2ee6cd5ef2083f4/packages/flutter/lib/src/material/icon_button.dart#L374). It seems that IconButton is missing `container: true`?

This was evident when we had an IconButton that was a child of a `scopesRoute: true, explicitChildNodes: true` Semantics node -- the IconButton wasn't showing up at all in semantic debugger.

Semantics widget seems to be required as a parent to the IconButton with a tooltip set which is used as the semantics label.

**Comment by nscobie** [comment]
Currently unable to reproduce, awaiting customer response on internal ticket.

Sample app for reference:

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

// Trying to repro https://github.com/flutter/flutter/issues/49119
class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      showSemanticsDebugger: true,
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: Scaffold(
        appBar: AppBar(title: Text('IconButton Testing')),
        body: GridView.count(
          crossAxisCount: 2,
          children: <Widget>[
            IconButton(
              icon: Icon(Icons.accessibility_new),
              onPressed: () => null,
              tooltip: 'accessibility',
            ),
            RawMaterialButton(
              child: Text('Button A'),
              onPressed: () => null,
            ),
            Semantics(
              scopesRoute: true,
              explicitChildNodes: true,
              child: Container(
                child: IconButton(
                  icon: Icon(Icons.access_alarm),
                  onPressed: () => null,
                  tooltip: 'alarm',
                ),
              ),
            ),
            Semantics(
              scopesRoute: true,
              explicitChildNodes: true,
              child: Container(
                child: RawMaterialButton(
                  child: Text('Button C'),
                  onPressed: () => null,
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

Output from `flutter doctor` for reference:
```
[✓] Flutter (Channel unknown, 1.20.0-3.0.pre.68, on Mac OS X 10.15.5 19F101, locale en)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
[✓] Android Studio (version 4.0)
[✓] VS Code (version 1.46.1)
[✓] Connected device (2 available)
```

**Comment by nscobie** [comment]
**Update:** received additional context of use case from customer. Unfortunately still unable to reproduce, asking for more follow up information.

_My response to the internal ticket:_

> My understanding is that IconButtons under ListTiles aren't showing up at all in the semantics debugger unless they are wrapped in a Semantics node with `container: true`? I've been trying many different configurations to try and replicate this behavior, but so far the only way I am able to make an IconButton not show up in the semantics debugger is by leaving `IconButton.tooltip` and `IconButton.onPressed` as null, and omitting a `semanticLabel` on the child Icon widget, which makes sense since there is no semantic information present. In this case, wrapping the IconButton in a Semantics node with `container: true` doesn't change the behavior.
> 
> **Follow up questions:**
> 
> 1. Which version of Flutter are you using?
> 2. Can you confirm the behavior on a physical device?
> 3. Are there any incorrect assumptions I'm making here, either with my understanding of the problem or with my attempted reproduction (see below)?
>     1. Can you provide a minimal reproduction which I could run?
> 
> Thanks!
> 
> **Attempted reproduction:**
> 
> ```dart
> import 'package:flutter/material.dart';
> 
> void main() {
>   runApp(MyApp());
> }
> 
> // Trying to repro https://github.com/flutter/flutter/issues/49119
> class MyApp extends StatelessWidget {
>   @override
>   Widget build(BuildContext context) {
>     return MaterialApp(
>       showSemanticsDebugger: true,
>       title: 'Flutter Demo',
>       theme: ThemeData(
>         primarySwatch: Colors.blue,
>         visualDensity: VisualDensity.adaptivePlatformDensity,
>       ),
>       home: Scaffold(
>         body: ListView(
>           children: [
>             Text('IconButtons w/ callbacks'),
>             ListTile(
>               leading: Icon(Icons.info),
>               title: Text('Plain IconButton'),
>               trailing: IconButton(
>                 icon: Icon(Icons.close),
>                 onPressed: () {},
>                 tooltip: 'CLOSE',
>               ),
>             ),
>             ListTile(
>               leading: Icon(Icons.info),
>               title: Text('Container IconButton'),
>               trailing: Semantics(
>                 container: true,
>                 child: IconButton(
>                   icon: Icon(Icons.close),
>                   onPressed: () {},
>                   tooltip: 'CLOSE',
>                 ),
>               ),
>             ),
>             ListTile(
>               leading: Icon(Icons.info),
>               title: Text('Plain IconButton w/ Icon label'),
>               trailing: IconButton(
>                 icon: Icon(Icons.close, semanticLabel: 'close icon'),
>                 onPressed: () {},
>                 tooltip: 'CLOSE',
>               ),
>             ),
>             ListTile(
>               leading: Icon(Icons.info),
>               title: Text('Container IconButton w/ Icon label'),
>               trailing: Semantics(
>                 container: true,
>                 child: IconButton(
>                   icon: Icon(Icons.close, semanticLabel: 'close icon'),
>                   onPressed: () {},
>                   tooltip: 'CLOSE',
>                 ),
>               ),
>             ),
> 
>             Text('IconButtons WITHOUT callbacks'),
>             ListTile(
>               leading: Icon(Icons.info),
>               title: Text('Plain IconButton'),
>               trailing: IconButton(
>                 icon: Icon(Icons.close),
>                 tooltip: 'CLOSE',
>               ),
>             ),
>             ListTile(
>               leading: Icon(Icons.info),
>               title: Text('Container IconButton'),
>               trailing: Semantics(
>                 container: true,
>                 child: IconButton(
>                   icon: Icon(Icons.close),
>                   tooltip: 'CLOSE',
>                 ),
>               ),
>             ),
>             ListTile(
>               leading: Icon(Icons.info),
>               title: Text('Plain IconButton w/ Icon label'),
>               trailing: IconButton(
>                 icon: Icon(Icons.close, semanticLabel: 'close icon'),
>                 tooltip: 'CLOSE',
>               ),
>             ),
>             ListTile(
>               leading: Icon(Icons.info),
>               title: Text('Container IconButton w/ Icon label'),
>               trailing: Semantics(
>                 container: true,
>                 child: IconButton(
>                   icon: Icon(Icons.close, semanticLabel: 'close icon'),
>                   tooltip: 'CLOSE',
>                 ),
>               ),
>             ),
> 
>             Text('IconButtons WITHOUT callbacks OR tooltips'),
>             ListTile(
>               leading: Icon(Icons.info),
>               title: Text('Plain IconButton'),
>               trailing: IconButton(
>                 icon: Icon(Icons.close),
>               ),
>             ),
>             ListTile(
>               leading: Icon(Icons.info),
>               title: Text('Container IconButton'),
>               trailing: Semantics(
>                 container: true,
>                 child: IconButton(
>                   icon: Icon(Icons.close),
>                 ),
>               ),
>             ),
>             ListTile(
>               leading: Icon(Icons.info),
>               title: Text('Plain IconButton w/ Icon label'),
>               trailing: IconButton(
>                 icon: Icon(Icons.close, semanticLabel: 'close icon'),
>               ),
>             ),
>             ListTile(
>               leading: Icon(Icons.info),
>               title: Text('Container IconButton w/ Icon label'),
>               trailing: Semantics(
>                 container: true,
>                 child: IconButton(
>                   icon: Icon(Icons.close, semanticLabel: 'close icon'),
>                 ),
>               ),
>             ),
>           ],
>         ),
>       ),
>     );
>   }
> }
> ```

**Comment by nscobie** [comment]
Didn't mean to let the bot close this. I unassigned myself due time constraints.

The customer has posted this follow-up, see b/147830016 for screenshot:

> The simplest reproduction of this issue is here:

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      showSemanticsDebugger: true, // Comment this line to see the visual layout.
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: Scaffold(
        backgroundColor: Colors.blue,
        appBar: AppBar(
          title: Text('IconButton semantics example'),
          backgroundColor: Colors.transparent,
          elevation: 0,
        ),
        body: Center(
          child: InkWell(
            onTap: () {},
            child: Row(children: [
              RaisedButton(child: Text('button')),
              Text('test'),
              IconButton(
                icon: const Icon(Icons.edit),
                tooltip: 'tooltip',
              ),
            ]),
          ),
        ),
      ),
    );
  }
}
```

>The IconButton doesn't show up as a button.
